### ISSUE TITLE
ESTree: Add ability to mark an `#[ast_meta]` added field as TS-only

### ISSUE THREAD

**Issue by overlookmotel** [oxc-project/oxc#9835]
From https://github.com/oxc-project/oxc/pull/9816#issue-2923808406:

> Adding back `null` causes a new mismatch due to ts-eslint has `ExpresssionStatement.directive: string | undefined`

Well that's really annoying! To achieve this, we need to add a "fake" `directive` field to `ExpresssionStatement` which is always `null`, and only appears in the TS AST.

The first part we can do, but making that field TS-only is currently not possible. Make it possible! Probably best way is to allow `#[ts]` attr on meta types.

**Comment by overlookmotel** [comment]
@hi-ogawa Do you see any better way to handle this? If not, I'll try and do this tomorrow.

**Comment by hi-ogawa** [comment]
Sounds good! With ts-only ast meta, I guess `ExpressionStatement` would look like this? 

```rs
#[estree(add_fields(directive = ExpressionStatementNull))]
pub struct ExpressionStatement<'a> { ... }

// serialize.rs
#[ts]
#[ast_meta]
#[estree(ts_type = "string | null")]
struct ExpressionStatementNull { ...}

impl Estree for ExpressionStatementNull { ... }
```

```ts
// types.d.ts
interface ExpressionStatement {
  ...,
  directive: string | null,
}
```

**Comment by overlookmotel** [comment]
Yes exactly. Except I'd call the "meta type" `ExpressionStatementDirectives`, to follow the existing naming convention.

I'll get on it now.

**Comment by overlookmotel** [comment]
Fixed in #9844.
